core/src/main/java/com/sequenceiq/cloudbreak/service/datalake/DiskUpdateService.java

```diff
@@ -0,0 +1,191 @@
+import com.sequenceiq.cloudbreak.orchestrator.state.ExitCriteriaModel;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterApiConnectors;
+import com.sequenceiq.cloudbreak.service.resource.ResourceService;
+import com.sequenceiq.cloudbreak.service.stack.StackDtoService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.common.api.type.ResourceType;
+
+@Service
+public class DiskUpdateService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DiskUpdateService.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private ClusterApiConnectors clusterApiConnectors;
+
+    @Inject
+    private StackUtil stackUtil;
+
+    @Inject
+    private StackDtoService stackDtoService;
+
+    @Inject
+    private CloudParameterCache cloudParameterCache;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private ResourceService resourceService;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private ResourceAttributeUtil resourceAttributeUtil;
+
+    public boolean isDiskTypeChangeSupported(String platform) {
+        return cloudParameterCache.isDiskTypeChangeSupported(platform);
+    }
+
+    public void updateDiskTypeAndSize(DiskUpdateRequest diskUpdateRequest, List<VolumeSetAttributes.Volume> volumesToUpdate,
+            String cloudPlatform) throws Exception {
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+        StackDto stackDto = stackDtoService.getById(diskUpdateRequest.getStackId());
+        AuthenticatedContext ac = getAuthenticatedContext(cloudConnector, stackDto);
+        List<String> volumeIds = volumesToUpdate.stream().map(VolumeSetAttributes.Volume::getId).toList();
+        cloudConnector.resources().updateDiskVolumes(ac, volumeIds, diskUpdateRequest.getVolumeType(), diskUpdateRequest.getSize());
+    }
+
+    public void stopCluster(long stackId) throws Exception {
+        StackDto stackDto = stackDtoService.getById(stackId);
+        ClusterApi clusterApi = clusterApiConnectors.getConnector(stackDto);
+        clusterApi.clusterModificationService().stopCluster(true);
+    }
+
+    public void startCluster(long stackId) throws Exception {
+        StackDto stackDto = stackDtoService.getById(stackId);
+        ClusterApi clusterApi = clusterApiConnectors.getConnector(stackDto);
+        clusterApi.clusterModificationService().startCluster();
+    }
+
+    public void resizeDisks(long stackId, String instanceGroup) throws Exception {
```

Comment:
split the method into more method, it is hard to read

Reply:
done added loggers as well

Reply:
thanks!